src/PureScript/Backend/Optimizer/Semantics.purs

```diff
@@ -773,6 +792,8 @@ evalPrimOpAssocL op match combine a b = case match a of
   _ ->
     Nothing
   where
+  a = deref a'
+  b = deref b'
```

Comment:
I don't think we want to do this here. This will apply to all associative operators, which doesn't make sense to me. I would expect this to be done before calling `evalPrimOpAssocL` in appropriate operators.

Reply:
I've moved this to do its thing in `evalPrimOpNum` and, for now, disabled it when doing string concatenation. This means, though, that my use case for Deku would no longer be covered, so I'd like to find a way to opt into string concatenation (I agree with you that it doesn't make sense to concatenate arbitrary strings).